src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -1053,6 +1068,190 @@ private void AddPrimarySidClaim(List<Claim> instanceClaims)
+                            Marshal.Copy(windowsClaim.Values.pUint64, uintValues, 0, (int)windowsClaim.ValueCount);
+
+                            for (int item = 0; item < windowsClaim.ValueCount; item++)
+                            {
+                                Claim c = new Claim(windowsClaim.Name, Convert.ToString((ulong)uintValues[item], CultureInfo.InvariantCulture), ClaimValueTypes.UInteger64, _issuerName, _issuerName, this);
+                                c.Properties.Add(propertyValue, string.Empty);
+                                instanceClaims.Add(c);
+                            }
+                            break;
+
+                        case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_BOOLEAN:
+                            long[] boolValues = new long[windowsClaim.ValueCount];
+                            Marshal.Copy(windowsClaim.Values.pUint64, boolValues, 0, (int)windowsClaim.ValueCount);
+
+                            for (int item = 0; item < windowsClaim.ValueCount; item++)
+                            {
+                                Claim c = new Claim(
+                                    windowsClaim.Name,
+                                    ((ulong)boolValues[item] == 0
+                                        ? Convert.ToString(false, CultureInfo.InvariantCulture)
+                                        : Convert.ToString(true, CultureInfo.InvariantCulture)),
+                                    ClaimValueTypes.Boolean,
+                                    _issuerName,
+                                    _issuerName,
+                                    this);
+
+                                c.Properties.Add(propertyValue, string.Empty);
+                                instanceClaims.Add(c);
+                            }
+                            break;
+
+
+                            // These claim types are defined in the structure found in winnt.h, but I haven't received confirmation (may  2011) that they are supported and are not enabled.
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_FQBN:
+                            //    break;
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_SID:
+                            //    break;
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_OCTET_STRING:
+                            //    break;
+
+                    }
+
+                    offset += Marshal.SizeOf(windowsClaim);
+                }
+            }
+            finally
+            {
+                safeAllocHandle.Close();
+            }
+        }
+
+        private static bool CheckForWindows8Properties()
+        {
+            using (WindowsIdentity tmp = GetCurrent(TokenAccessLevels.Read))
+            using (SafeLocalAllocHandle safeLocalAllocHandle = new SafeLocalAllocHandle(IntPtr.Zero))
```

Comment:
Is it possible any of this could throw, which would result in subsequent attempts to access the lazy.Value throwing?

Reply:
It shouldn't.  But I changed it to a different style of lazy (record first failure) so now it's just more in-line and less susceptible to a noisy oops.